### PR TITLE
Make the weekly.yml link in lts-backports-changelog optional + Support HTTPS URLs as a source of weekly.yml in lts-backports-changelog+ Fix failures of lts-backports-changelog on platforms where bash is not a default shell

### PR DESCRIPTION
**Pull request by oleg-nenashev**
Some fixes which actually make ` lts-backports-changelog` operational in Docker

- [x] Make the weekly.yml link in lts-backports-changelog optional 
- [x] Support HTTPS URLs as a source of weekly.yml in lts-backports-changelog
- [x] Fix failures of lts-backports-changelog on platforms where bash is not a default shell (like dash in Docker)

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
This pull request title makes me 😢

File: lts-backports-changelog.rb [hunk 1]
```diff
@@ -1,19 +1,24 @@
...
 curl_auth = ENV['JIRA_AUTH']
 
-if ARGV.length != 2
-	puts "Usage:    generate-lts-changelog.rb <LTS version> <weekly.yml>"
+if ARGV.length < 1 || ARGV.length > 2
+	puts "Usage:    generate-lts-changelog.rb <LTS version> [weekly.yml]"
+	puts ""
+	puts "Default weekly.yml: https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml"
 	puts ""
 	puts "Missing argument <version> and/or <weekly.yml>"
```
[the remark this answers is not in the document]
Comment (oleg-nenashev) [reply lts-backports-changelog.rb]:
```suggestion
	puts "ERROR: Wrong argument number"
```

File: lts-backports-changelog.rb [hunk 2]
```diff
@@ -22,11 +27,23 @@
 
 weekly_changelog_file = ARGV[1]
 
-backports_str = `set -o pipefail ; curl --fail -u #{curl_auth} -X POST --data '{"startAt":0, "maxResults":1000,"fields": ["key"], "jql": "labels = #{lts}-fixed" }' -H "Content-Type: application/json" https://issues.jenkins-ci.org/rest/api/2/search | jq -r '.issues[] | .key'`
+command = "set -o pipefail ; curl --fail -u #{curl_auth} -X POST --data '{\"startAt\":0, \"maxResults\":1000,\"fields\": [\"key\"], \"jql\": \"labels = #{lts}-fixed\" }' -H \"Content-Type: application/json\" https://issues.jenkins-ci.org/rest/api/2/search | jq -r '.issues[] | .key'"
```
Comment (MarkEWaite) [inline lts-backports-changelog.rb]: I had to remove the `set -o pipefail` in my usage of the code, but did not spend time to investigate why my environment is different than other environments.  I was running on a Ubuntu 18.04 Linux machine (if that provides any hints why my machine behaved differently).  I believe I was using the program outside a docker container, not running it from a docker container.

No objection from me if this is left exactly as it is.  I will eventually make time enough to diagnose why my machine could not call `set -o pipefail`.

```suggestion
command = "set -o pipefail ; curl --fail -u #{curl_auth} -X POST --data '{\"startAt\":0, \"maxResults\":1000,\"fields\": [\"key\"], \"jql\": \"labels = #{lts}-fixed\" }' -H \"Content-Type: application/json\" https://issues.jenkins-ci.org/rest/api/2/search | jq -r '.issues[] | .key'"
```

Comment (oleg-nenashev) [reply lts-backports-changelog.rb]: > No objection from me if this is left exactly as it is. I will eventually make time enough to diagnose why my machine could not call set -o pipefail.

Likely your default shell is not Bash

**Pull request merged**